### PR TITLE
Rust - use serde

### PR DESCRIPTION
**Pull request by notoriaga**
Little bit of a different approach. There is quite a bit of inconstancy in the setttings.yaml, which is pretty easy to handle with serde but maybe might be worth just changing the settings? Not sure if other things are dependent on it though.

### CODE REVIEW

File: rust/src/settings_manager.rs [hunk 1]
```diff
@@ -1,72 +1,133 @@
...
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Setting {
+    name: String,
+
+    group: String,
+
+    #[serde(rename = "type")]
+    kind: Option<SettingKind>,
```
Comment (notoriaga) [inline rust/src/settings_manager.rs:22]: Some settings in group cn0_est have no type, I feel like at the very least we should add those in

File: rust/src/settings_manager.rs [hunk 2]
```diff
@@ -1,72 +1,133 @@
...
 
-use yaml_rust::{Yaml, YamlLoader};
+use serde::{
+    de::{self, Unexpected},
+    Deserialize,
+};
 
-const SOURCE: &str = include_str!("../../settings.yaml");
+lazy_static::lazy_static! {
+    static ref SETTINGS: Vec<Setting> = {
```
Comment (notoriaga) [inline rust/src/settings_manager.rs:9]: Originally the settings were wrapped in a mutex, which we don't need if this is going to be readonly (which I think it is)

File: rust/src/settings_manager.rs [hunk 3]
```diff
@@ -1,72 +1,133 @@
...
+    Double,
+
+    #[serde(rename = "string")]
+    String,
+
+    #[serde(rename = "enum")]
+    Enum,
+
+    #[serde(rename = "packed bitfield")]
+    PackedBitfield,
```
Comment (notoriaga) [inline rust/src/settings_manager.rs:68]: There is only one setting with this type, not sure the best way to handle it, maybe just as a string?

File: rust/src/settings_manager.rs (hunk 2 above)
Comment (silverjam) [reply rust/src/settings_manager.rs:9]: 👍

File: rust/src/settings_manager.rs (hunk 1 above)
Comment (silverjam) [reply rust/src/settings_manager.rs:22]: Yes, we should

File: rust/src/settings_manager.rs (hunk 3 above)
Comment (silverjam) [reply rust/src/settings_manager.rs:68]: String is probably fine

Comment (silverjam) [reply rust/src/settings_manager.rs:68]: (The setting will get written/read as a string anyway so as long as we are able to preserve all the data then a String should be fine)

**Comment by silverjam** [comment]
> Little bit of a different approach. There is quite a bit of inconstancy in the setttings.yaml, which is pretty easy to handle with serde but maybe might be worth just changing the settings? Not sure if other things are dependent on it though.

Only thing that will be dependent is the console (piksi_tools) as far as I know and I think it handles the inconsistency either through the YAML parser or manually, so I can go either way on this, the code to just handle the inconsistency looks good and it'll give us the ability to understand process old settings YAMLs that haven't been cleaned up (not sure we'll ever need this, but it seems useful).  If we find that handling the inconsistency becomes a maintenance burden, we should abandon it then.

**Review by silverjam — APPROVED** [review]

File: rust/src/settings_manager.rs (hunk 1 above)
Comment (notoriaga) [reply rust/src/settings_manager.rs:22]: These are the fields without types, any idea who we should ping to get the values?

```
- group: cn0_est
  name: alpha
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: nbw
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: scale
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cn0 shift
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cn0_shift
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cutoff
  expert: true
  Description: cn0 cutoff.
  readonly: false
```

Comment (silverjam) [reply rust/src/settings_manager.rs:22]: > 
> 
> These are the fields without types, any idea who we should ping to get the values?
> 
> ```
> - group: cn0_est
>   name: alpha
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: nbw
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: scale
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cn0 shift
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cn0_shift
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cutoff
>   expert: true
>   Description: cn0 cutoff.
>   readonly: false
> ```

These don't seem to exist in any code anywhere anymore: https://github.com/search?q=org%3Aswift-nav+cn0_est&type=code -- so we can either remove or ignore these

**Pull request merged**